examples/gno.land/p/gov/dao/types.gno

```diff
@@ -0,0 +1,43 @@
+package dao
+
+import (
+	"std"
+
+	"gno.land/p/gov/proposal"
+)
+
+type VoteOption string
+
+const (
+	YesVote VoteOption = "YES"
+	NoVote  VoteOption = "NO"
+)
+
+func (v VoteOption) String() string {
+	return string(v)
+}
+
+// Vote encompasses a single GOVDAO vote
+type Vote struct {
+	Address std.Address // the address of the voter
+	Option  VoteOption  // the voting option
+}
+
+// DAO defines the DAO abstraction
+type DAO interface {
+	// Propose adds a new proposal to the executor-based GOVDAO.
+	// Returns the generated proposal ID
+	Propose(comment string, executor proposal.Executor) (uint64, error)
+
+	// VoteOnProposal adds a vote to the given proposal ID
+	VoteOnProposal(id uint64, option VoteOption) error
+
+	// ExecuteProposal executes the proposal with the given ID
+	ExecuteProposal(id uint64) error
+
+	// GetMembStore returns the member store associated with the DAO
+	GetMembStore() MemberStore
+
+	// GetPropStore returns the proposal store associated with the DAO
+	GetPropStore() PropStore
```

Comment:
I think it's not an idea to have independent interfaces since they are tightly coupled to this top-level interface. I understand your preference for splitting things, but in this case, it's about making the DAO parameters unavailable from the MembStore and PrepStore, which is unfortunate because most of your helpers in the stores are checking if the caller is the DAO.

I suggest embedding (or flattening) the interfaces in the main interface instead of returning an independent interface. This way, you can have a single struct implementing everything or decide to use embedded standalone MemberStore/PropStore in the implementation struct.

Reply:
There is tight coupling with what the top level DAO API does, I agree 🙁 

I wanted to preserve the shallow API for the DAO, but also not limit the functionality so we can create wrapper Realms to do much more sophisticated and complicated queries (ex. return JSON objects...)

I've definitely thought about embedding the interfaces in `DAO`, but I'm struggling to understand how this is a better option than having 2 getters, if we want to use a `DAO` interface in `/r/gov/dao`. What if there is a DAO implementation that doesn't publicly expose changing the member set (ex. returns `nil` for `GetMembStore()`)?

Do we and pipe relevant methods, and let the caller do whatever they want with the `DAO` through a fetcher?
What I'm essentially asking is, does our `r/gov/dao` realm look like this?

```go
package dao

var d dao.DAO

func init() {
	d := simpledao.New(...)
}

func Propose(comment string, executor proposal.Executor) (uint64, error) {
	return d.Propose(comment, executor)
}

func VoteOnProposal(id uint64, option VoteOption) error {
	return d.Propose(id, option)
}

func ExecuteProposal(id uint64) error {
	return d.Propose(id)
}

func GetDAO() dao.DAO {
	return d
}

func Render(path string) string {
	// ...
}

```

Reply:
Through this API, `r/gov/dao` is minimal, and we don't have a gazillion methods in our public API, but let the caller wrap them through another Realm that calls `GetDAO`

I just need clarify whether we need this:
>What if there is a DAO implementation that doesn't publicly expose changing the member set (ex. returns nil for GetMembStore())?

- If the answer is yes, I'd prefer to keep the `GetXXX` on the `DAO` interface, and not embed
- If the answer is no, I see no reason for not embedding

Reply:
You’re right.

Let’s remove the notion of member from p/gov/dao so that a DAO is an object that:
- handles proposals
- have assets

member management is something that can be made generic but it’s not because of a Dao, it’s just for anything that wants members.

r/gov/dao should implements p/gov/dao (the proposal framework) AND p/something/membership; while relying on the proposal framework (injecting it?) to manage the members through proposals.

Reply:
I've applied this change, but can't find the exact commit 🤦‍♂️